### PR TITLE
Charity payment

### PR DESCRIPTION
**Pull request by aabounegm**
Implements the payment flow for charities.

One known bug regarding login is that every time the web app is closed and reopened, our cookies seem to get cleared and the access token is lost, but clicking "login" immediately authenticates with the _sudir_ portal. I will explore this issue in another pull request.

### CODE REVIEW

File: src/lib/api/mts.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const API_BASE = 'https://test.mospay.mos.ru/mospaynew';
+export const getAccessTokenUrl = (redirectUrl: string) => {
+	const AUTH_URL_BASE = 'https://login.mos.ru/sps/oauth/ae';
+	const params = new URLSearchParams({
+		scope: 'openid profile',
+		response_type: 'code',
+		redirect_uri: 'https://test.mospay.mos.ru/sudir/auth',
```
Comment (GuVictory) [inline src/lib/api/mts.ts]: Lets move it to base url

File: src/routes/charities/+layout.server.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 
 export const load: LayoutServerLoad = async ({ fetch, cookies, url }) => {
 	let accessToken = url.searchParams.get('accessToken') ?? cookies.get('accessToken');
 	if (accessToken == undefined) {
 		// No access token found. Go to the MTS login page, and redirect to the same page again
 		// SvelteKit seems to default it to HTTP during dev
-		const redirectUrl = getAccessTokenUrl(url.href.replace('http:', 'https:'));
-		throw redirect(307, redirectUrl);
+		const redirectUrl = url.href.replace('http:', 'https:');
+		throw redirect(307, `/login?redirectUrl=${redirectUrl}`);
```
Comment (GuVictory) [inline src/routes/charities/+layout.server.ts:12]: Lets use `URLSearchParams` here?

File: src/routes/charities/+layout.server.ts [hunk 3]
```diff
@@ -18,7 +19,8 @@ export const load: LayoutServerLoad = async ({ fetch, cookies, url }) => {
 	if (!res.ok) {
 		if (res.status === 401) {
 			// Token expired. Redirect to MTS login to renew
-			throw redirect(307, getAccessTokenUrl(url.href));
+			const redirectUrl = url.href.replace('http:', 'https:');
+			throw redirect(307, `/login?redirectUrl=${redirectUrl}`);
```
Comment (GuVictory) [inline src/routes/charities/+layout.server.ts:23]: same

**Review by GuVictory — APPROVED** [review]

File: src/routes/charities/+layout.server.ts (hunk 2 above)
Comment (aabounegm) [reply src/routes/charities/+layout.server.ts:12]: I don't think it's worth it for such a small URL, it's just one parameter

**Pull request merged**